react-app/src/components/HomeScreen/HomeScreen.tsx

```diff
@@ -1,15 +1,35 @@
-import React from "react";
+import React, { useCallback } from "react";
 import cn from "classnames";
 import { Link } from "react-router-dom";
 import AppRoutes from "../../navigation/AppRoutes";
 import LocalizedText from "../Localized/LocalizedText";
+import { useLocale } from "../../providers/AppLocaleProvider";
+import { Locale } from "../../i18n/LocaleModel";
 
 const HomeScreen: React.FC = () => {
+  const { setLocale: setLocale_ } = useLocale();
+
+  const setLocale = useCallback(
```

Comment:
onClick callback should handle React.MouseEvent...
please refactor the button component?